plugin/pulpcore/plugin/stages/association_stages.py

```diff
@@ -0,0 +1,158 @@
+                to be associated.
+            out_q (:class:`asyncio.Queue`): Each item is a :class:`django.db.models.query.QuerySet`
+                of :class:`~pulpcore.plugin.models.Content` subclass that are already associated but
+                not included in the stream of items from `in_q`. One
+                :class:`django.db.models.query.QuerySet` is put for each
+                :class:`~pulpcore.plugin.models.Content` type.
+        """
+        with ProgressBar(message='Associating Content') as pb:
+            declarative_content_list = []
+            shutdown = False
+            while True:
+                try:
+                    declarative_content = in_q.get_nowait()
+                except asyncio.QueueEmpty:
+                    if not declarative_content_list and not shutdown:
+                        declarative_content = await in_q.get()
+                        declarative_content_list.append(declarative_content)
+                        continue
+                else:
+                    declarative_content_list.append(declarative_content)
+                    continue
+
+                content_q_by_type = defaultdict(lambda: Q(pk=None))
+                for declarative_content in declarative_content_list:
+                    if declarative_content is None:
+                        shutdown = True
+                        continue
+                    try:
+                        unit_key = declarative_content.content.natural_key()
+                        self.unit_keys_by_type[type(declarative_content.content)].remove(unit_key)
+                    except KeyError:
+                        model_type = type(declarative_content.content)
+                        unit_key_dict = declarative_content.content.natural_key_dict()
+                        unit_q = Q(**unit_key_dict)
+                        content_q_by_type[model_type] = content_q_by_type[model_type] | unit_q
+
+                for model_type, q_object in content_q_by_type.items():
+                    queryset = model_type.objects.filter(q_object)
+                    self.new_version.add_content(queryset)
+                    pb.done = pb.done + queryset.count()
+                    pb.save()
+
+                if shutdown:
+                    break
+                declarative_content_list = []
+
+            for unit_type, ids in self.unit_keys_by_type.items():
+                if ids:
+                    units_to_unassociate = Q()
+                    for unit_key in self.unit_keys_by_type[unit_type]:
+                        query_dict = {}
+                        for i, key_name in enumerate(unit_type.natural_key_fields()):
+                            query_dict[key_name] = unit_key[i]
+                        units_to_unassociate |= Q(**query_dict)
+                    await out_q.put(unit_type.objects.filter(units_to_unassociate))
+            await out_q.put(None)
+
+
+class content_unit_unassociation:
```

Comment:
Class name should be CapWords - PEP8.

Reply:
Same question as [here](https://github.com/pulp/pulp/pull/3559#discussion_r206939231).